### ISSUE TITLE
io.open and mmap

### ISSUE THREAD

**Issue by ironpythonbot** [IronLanguages/main#1081]
``` python
with io.open("somefile", "r+b") as f:
        mm = mmap.mmap(f.fileno(), 0)
```

fails with: 

``` python
mmap.error: [Errno 9] Bad file descriptor  
```
## Work Item Details

**Original CodePlex Issue:**    [Issue 34528](http://ironpython.codeplex.com/workitem/34528)
**Status:** Proposed
**Reason Closed:**  Unassigned
**Assigned to:**    Unassigned
**Reported on:**    Oct 3, 2013 at 3:34 PM
**Reported by:**    Libbnor / @slozier 
**Updated on:** Oct 3, 2013 at 3:34 PM
**Updated by:** Libbnor

**Comment by slozier** [comment]
Just marking this one as reported by me.

**Comment by simplicbe** [comment]
@slozier the problem seems to be in `fileno()` which always returns -1.

**Comment by paweljasinski** [comment]
@slozier can you try open instead of io.open?

**Comment by slozier** [comment]
@paweljasinski open works fine.

I guess the issue should probably be that when using io.open the fileno method doesn't return a usable file descriptor.

**Comment by simplicbe** [comment]
Yes, `io.fileno` can't work, because no file which will be opened are registered in the `PythonFileManager`. An there are more things missing.

As an additional information:

```
with io.open("somefile", "r+b") as f:
        mm = mmap.mmap(f.fileno(), 0)
```

Can't work on windows machine anyway, because the size `0` is not allowed.

**Comment by slozier** [comment]
Using 0 as the length argument means use the size of the file.

**Comment by simplicbe** [comment]
Than there is something wrong in the implementation of `mmap` in `mmap.cs`

**Comment by slozier** [comment]
Yes, there appears to be something wrong with the implementation of `mmap`. The length of the map is too big when using `0` as the length argument. The following sample outputs 4096 (where test1081.py is the sample):

```
import mmap
with open("test1081.py", "r+b") as f:
    mm = mmap.mmap(f.fileno(), 0)
    print len(mm)
```

Edit: the length issue above is fixed with #1519
